subsys/bluetooth/host/iso.c

```diff
@@ -1383,7 +1398,21 @@ int bt_iso_big_terminate(struct bt_iso_big *big)
 void hci_le_big_complete(struct net_buf *buf)
 {
 	struct bt_hci_evt_le_big_complete *evt = (void *)buf->data;
-	struct bt_iso_big *big = &bigs[evt->big_handle];
+	struct bt_iso_big *big;
+
+	if (!(evt->big_handle < ARRAY_SIZE(bigs))) {
+		BT_WARN("Invalid BIG handle");
+
+		big = big_lookup_flag(BT_BIG_PENDING);
+		if (big) {
+			cleanup_big(big);
```

Comment:
I think we need to do the same here as we do for per-adv and call the disconnected callbacks by calling `big_disconnect` before `cleanup_big`.

Reply:
I'm unsure about the `bit_disconnect` you mentioned, if you see below here other if the number of BIS is invalid there is also just cleanup_big.

Is this correct?

Reply:
I think that's actually rather a missing `big_disconnect(big);` from below. I guess I was unsure whether or not to call the connected/disconnect callbacks for the broadcaster, but since we are also calling `bt_conn_set_state(bis->conn, BT_CONN_CONNECTED);` (which calls the `connected` callback), then we should also call the `disconnected` callback. 

So I suggest adding it to both places.